File: stablecoin-blocklist-events-js/src/storage.js
const { fetchJwt } = require("forta-agent");
const { readFileSync } = require("fs");
const fetch = require("node-fetch");

const owner_db = "https://research.forta.network/database/owner/";

const testMode =
  process.env.NODE_ENV && process.env.NODE_ENV.includes("production") ? "main" : "test";

async function _token() {
  const tk = await fetchJwt();
  return { Authorization: `Bearer ${tk}` };
}

async function _loadJson(key) {
  if (testMode === "test") {
    // Loading JSON from local file secrets.json
    const json = readFileSync("secrets.json");
    return JSON.parse(json);
  } else {
    return await fetch(`${owner_db}${key}`, { headers: _token() })
      .then(response => {
        if (response.ok) {
          return response.json();
        } else {
          throw new Error(
            `Error loading JSON from owner db: ${response.status}, ${response.statusText}`
          );
        }
      })
      .catch(error => {
        throw new Error(`Error loading JSON from owner db: ${error.message}`);
      });
  }
}

async function getSecrets() {
  return await _loadJson("secrets.json");
}

module.exports = { getSecrets };

File: early-attack-detector-py/src/storage.py
import forta_bot_sdk
import json
import aiohttp
import os

owner_db = "https://research.forta.network/database/owner/"

test_mode = "main" if 'FORTA_ENV' in os.environ and 'production' in os.environ.get(
    'FORTA_ENV') else "test"

async def _token():
    tk = await forta_bot_sdk.fetch_jwt()
    return {"Authorization": f"Bearer {tk}"}

async def _load_json(key: str) -> object:
    if test_mode == "test":
        with open("secrets.json") as f:
            return json.load(f)
    else:
        async with aiohttp.ClientSession() as session:
            async with session.get(f"{owner_db}{key}", headers=await _token()) as res:
                if res.status == 200:
                    try:
                        # Attempt to parse the response as JSON regardless of the Content-Type
                        return await res.json(content_type=None)
                    except json.JSONDecodeError:
                        raise Exception("Failed to decode JSON response")
                else:
                    raise Exception(
                        f"error loading json from owner db: {res.status}, {await res.text()}")


async def get_secrets():
    return await _load_json("secrets.json")
